packages/server/graphql/private/mutations/startTrial.ts

```diff
@@ -0,0 +1,46 @@
+import removeTeamsLimitObjects from '../../../billing/helpers/removeTeamsLimitObjects'
+import getRethink from '../../../database/rethinkDriver'
+import getKysely from '../../../postgres/getKysely'
+import setTierForOrgUsers from '../../../utils/setTierForOrgUsers'
+import setUserTierForOrgId from '../../../utils/setUserTierForOrgId'
+import hideConversionModal from '../../mutations/helpers/hideConversionModal'
+import {MutationResolvers} from '../resolverTypes'
+
+const startTrial: MutationResolvers['startTrial'] = async (_source, {orgId}, {dataLoader}) => {
+  const r = await getRethink()
+  const pg = getKysely()
+  const now = new Date()
+  const organization = await dataLoader.get('organizations').load(orgId)
+
+  // VALIDATION
+  if (organization.tier !== 'starter') {
+    throw new Error('Cannot start trial for organization on paid tier')
+  }
+  if (organization.trialStartDate) {
+    throw new Error(`Trial already started for org. Start date: ${organization.trialStartDate}`)
+  }
+
+  // RESOLUTION
+  await Promise.all([
+    r({
+      updatedOrg: r.table('Organization').get(orgId).update({
+        trialStartDate: now,
+        tierLimitExceededAt: null,
+        scheduledLockAt: null,
+        lockedAt: null,
+        updatedAt: now
+      })
+    }).run(),
+    pg.updateTable('Team').set({trialStartDate: now}).where('orgId', '=', orgId).execute(),
+    removeTeamsLimitObjects(orgId, dataLoader)
+  ])
+  organization.trialStartDate = now
+
+  await Promise.all([setUserTierForOrgId(orgId), setTierForOrgUsers(orgId)])
```

Comment:
+1 it's not obvious that `setUserTierForOrgId` and `setTierForOrgUsers` can run in parallel as one writes `OrganizationUser` and the other one reads it. A comment on top of `setUserTierForUserIds` that it does not read the `tier` from `OrganizationUser` for this very reason would be nice.

Reply:
Added a comment.